ios/FluentUI/Vnext/Avatar/AvatarGroup.swift

```diff
@@ -0,0 +1,191 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+@objc public enum MSFAvatarGroupStyle: Int, CaseIterable {
+    case stack
+    case pile
+}
+//
+/// `MSFAvatarGroupState` defines properties in an AvatarGroup
+///
+/// `maxDisplayedAvatars`: Maximum number of avatars displayed
+///
+/// `overflowCount`: Number to be displayed in the overflow avatar
+///
+/// `style`: Type of AvatarGroup. Avatars in a Stack is layered on top of each other; Pile separates avatars
+///
+/// `createAvatar`: Create an Avatar using an AvatarStyle and AvatarSize
+///
+/// `getAvatarState`: Get the state of an Avatar using an index value
+///
+/// `deleteAvatar`: Delete an Avatar with a specified index value
+@objc public protocol MSFAvatarGroupState {
+    var maxDisplayedAvatars: Int { get set }
+    var overflowCount: Int { get set }
+    var style: MSFAvatarGroupStyle { get set }
+
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize)
+    func getAvatarState(_ atIndex: Int) -> MSFAvatarState?
+    func deleteAvatar(atIndex: Int)
+}
+
+/// Properties that make up AvatarGroup content
+class MSFAvatarGroupStateImpl: NSObject, ObservableObject, MSFAvatarGroupState {
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize) {
+        avatars.append(AvatarView(style: style, size: size))
+    }
+
+    func getAvatarState(_ atIndex: Int) -> MSFAvatarState? {
+        if atIndex >= avatars.count {
```

Comment:
Nit: might be a good place for a `guard` statement (same would be true for `deleteAvatar` below)